File: test/laz_reader_test.cpp
#include <catch2/catch.hpp>
#include <cmath>
#include <copc-lib/io/laz_reader.hpp>
#include <fstream>
#include <limits>

using namespace copc;
using namespace copc::laz;
using namespace std;

TEST_CASE("LazReader tests", "[LazReader]")
{
    GIVEN("A valid file path")
    {

        LazFileReader reader("autzen-classified.copc.laz");

        SECTION("GetHeader Test")
        {
            auto header = reader.LazConfig().LasHeader();
            REQUIRE(header.PointFormatId() == 7);
            REQUIRE(header.PointCount() == 10653336);
            REQUIRE(header.PointRecordLength() == 36);
            REQUIRE(header.EbByteSize() == 0);
        }

        SECTION("WKT")
        {
            auto wkt = reader.LazConfig().Wkt();
            REQUIRE(wkt.empty());
        }
    }

    GIVEN("An invalid file path") { REQUIRE_THROWS(LazFileReader("invalid_path/non_existant_file.copc.laz")); }

    GIVEN("A valid input stream")
    {
        fstream in_stream;
        in_stream.open("autzen-classified.copc.laz", ios::in | ios::binary);

        LazReader reader(&in_stream);

        SECTION("GetHeader Test")
        {
            auto header = reader.LazConfig().LasHeader();
            REQUIRE(header.PointFormatId() == 7);
            REQUIRE(header.PointCount() == 10653336);
        }

        SECTION("WKT")
        {
            auto wkt = reader.LazConfig().Wkt();
            REQUIRE(wkt.empty());
        }
    }
}

TEST_CASE("LazReader GetExtraByteVlrs Test", "[LazReader]")
{
    GIVEN("A valid file path")
    {
        LazFileReader reader("autzen-classified.copc.laz");

        auto eb_vlr = reader.LazConfig().ExtraBytesVlr();
        REQUIRE(eb_vlr.items.empty());
    }
}

File: test/laz_reader_test.py
import copclib as copc
import pytest

from .utils import get_autzen_file


def test_reader():
    # Given a valid file path
    reader = copc.LazReader(get_autzen_file())

    # Given an invalid file path
    with pytest.raises(RuntimeError):
        assert copc.LazReader("invalid_path/non_existant_file.copc.laz")

    # LasHeader Test
    header = reader.laz_config.las_header
    assert header.point_format_id == 7
    assert header.point_count == 10653336
    assert header.point_record_length == 36
    assert header.EbByteSize() == 0

    # WKT Test
    wkt = reader.laz_config.wkt
    assert len(wkt) == 0
